pkg/apis/extensions/validation/validation.go

```diff
@@ -346,15 +346,20 @@ func ValidateJobSpec(spec *extensions.JobSpec) errs.ValidationErrorList {
 		allErrs = append(allErrs, errs.NewFieldInvalid("completions", spec.Completions, isNegativeErrorMsg))
 	}
 
-	selector := labels.Set(spec.Selector).AsSelector()
-	if selector.Empty() {
-		allErrs = append(allErrs, errs.NewFieldRequired("selector"))
+	if len(spec.Selector.MatchLabels)+len(spec.Selector.MatchExpressions) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("(matchLabels or matchExpressions)"))
```

Comment:
FWIW, only a single field can be specified to NewFieldRequired. More details can be added elsewhere in the status response.

Here's what we did for generateName:
https://github.com/kubernetes/kubernetes/blob/6217869085aaf795a425da5a2b08742a2e6560e0/pkg/api/validation/validation.go#L245

Anyway, this will be fixed if we just require a non-nil Selector instead.